### PR TITLE
React style guide

### PR DESCRIPTION
**Pull request by Portals**
Here are some of the things I want to add before merging:

- [x]  An example on how the file structure should look like
- [x] Always prefer using stateless components. 
- [x] Avoid wrapper hell
- [x] Add .prettierrc file 
- [x] Add information about /api folder
- [x] If a view needs views, and that view is the only one needing those views. Then the folder should be named sub-views.

### CODE REVIEW

**Comment by Hultner** [comment]
You do know there's native HR's in markdown

---
Like this `---` ^

**Comment by Portals** [comment]
Did not know, will fix! Thanks!

**Comment by Portals** [comment]
I have updated my todo list with the following: Always try to use stateless components and that you should write good tests.

File: react/README.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+If you're unsure about the tools used or ReactJS generally, then you shouldn't be the one setting up the project or initalizing the different tools. Ask someone who has experince. It's often easier to see how someone else does something and try to mimic that instead of starting on your own and trying to follow this styleguide.
+
+---
+
+## Different kinds of components
+
+### Element
+
+An element is a stateless and [pure](https://en.wikipedia.org/wiki/Pure_function) component that should only focus on presentational matters. They should take in props, and display it. An element may contain other elements, but not other views, screens or use-cases. An element should not use any hooks since it's [pure](https://en.wikipedia.org/wiki/Pure_function). It should always to a functional component, not a class component.
```
Comment (Hoidi) [inline react/README.md]: Last sentence. It should always what? Is something missing or am I missing something?

File: react/README.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+### File extension
+
+The file extension should either be .json, .jsx or .js. .js should only be used for index.js and hooks. .json should only be used for translations. .jsx should be used for the rest.
+
+---
+
+## Folder structure
+
+Within the src folder, there should only be three folders. [Common](#common), [Common-ui](#common-ui), [Use-cases](#use-case) and [App](#app).
```
Comment (Hoidi) [inline react/README.md]: I count four folder. And what about api?

**Review by Hoidi — CHANGES_REQUESTED** [review]

**Comment by Portals** [comment]
@Hoidi Resolved your issues

**Review by Hoidi — APPROVED** [review]

**Review by Riddarvid — APPROVED** [review]

**Review by Flakt — APPROVED** [review]

**Review by cajoho99 — APPROVED** [review]

**Review by lindskogen — APPROVED** [review]

**Review by Kalior — APPROVED** [review]

**Review by EmElw — APPROVED** [review]

**Review by LatteG — APPROVED** [review]

**Pull request merged**